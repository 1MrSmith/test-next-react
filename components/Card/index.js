import React, { Fragment, memo, } from 'react';
import Link from 'next/link';
import { useAppContext } from '../AppContext';
import asyncLocalStorage from '../../services/local-storage.service';
import addToBookmark from '../../utils/add-to-bookmark';
import config from '../../config';
import style from './style.module.scss';

const Card = (props) => {
  const { userSearchString } = useAppContext();

  const cleanUp = () => {
    userSearchString('');
  }

  const onChange = async (event) => {

    const id = event.target.name;
    const userStorage = await asyncLocalStorage.getItem(config.keyLocalStorage);
    const user = JSON.parse(userStorage);

    const updatedUser = addToBookmark(user, props.typeVideo, id);

    await asyncLocalStorage.setItem(config.keyLocalStorage, JSON.stringify(updatedUser));

  }

  return (
    <Fragment>
      <Link
      className={style['section-card']}
      onClick={cleanUp}
      href={`${props.detailsUrl}/${props.id}`}>

        <div
        className={style['section-card__wrapper']}>

          <img
          className={style['section-card__img']}
          src={props.imageLink ? props.imageLink : ''}
          alt={props.title} />

        </div>

      </Link>
      <span
      className={style['bookmark']}>

        <input
        type='checkbox'
        className={style['bookmark-input']}
        name={props.id}
        id={props.id}
        defaultChecked={props.checked}
        onClick={onChange} />

        <label
        htmlFor={props.id}
        className={style['bookmark-star']}></label>

      </span>
    </Fragment>
  );
}

export default memo(Card);
