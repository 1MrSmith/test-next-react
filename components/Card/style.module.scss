.section-card {
  width: 100%;
  display: flex;
  flex-direction: column;
  border-radius: 0.6rem;
  box-shadow: 0 0.2rem 1.6rem 0 rgba(0, 0, 0, 0.1);
  background-size: cover;

  &__wrapper {
    position: relative;
  }

  &__btn {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    background-color: #555;
    color: white;
    padding: 12px 24px;
    border: none;
    border-radius: 5px;
  }

  &__img {
    width: 100%;
    height: 18.1rem;
    display: flex;
    justify-content: space-between;
    border-radius: 0.6rem;
    background-size: cover;
  }
}


.bookmark {
  display: inline-block;
  position: absolute;
  top: 10%;
  left: 80%;
  transform: translate(-10%, -80%);
}
.bookmark-input {
  margin: 0 0 0 -1rem;
  opacity: 0;
}

.bookmark:hover .bookmark-star:hover,
.bookmark:hover .bookmark-star:hover ~ .bookmark-star,
.bookmark-input:checked ~ .bookmark-star {
  background-position: 0 0;
}

.bookmark-star,
.bookmark:hover .bookmark-star {
  position: relative;
  display: block;
  cursor: pointer;
  width: 1rem;
  height: 1rem;
  background: url('../../assets/stars.png') 0 -1rem;
}

