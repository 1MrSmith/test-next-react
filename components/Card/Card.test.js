import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Card from './index';

describe('Card component', () => {
  let realUseContext;
  let useContextMock;
  let stringMock, numberMock, booleanMock;

  beforeEach(() => {
    stringMock = '';
    numberMock = 1;
    booleanMock = false;
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
  });

  afterEach(() => {
    React.useContext = realUseContext;
  });

  it('renders without crashing', async() => {
    useContextMock.mockReturnValue('Test Value');

    const element = new ShallowRenderer().render(
      <Card
        key={numberMock}
        id={numberMock}
        title={stringMock}
        imageLink={stringMock}
        detailsUrl={stringMock}
        typeVideo={stringMock}
        checked={booleanMock} />
    );

    expect(element).toMatchSnapshot();

  });

});
