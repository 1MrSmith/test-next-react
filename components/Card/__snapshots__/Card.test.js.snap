// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders without crashing 1`] = `
<React.Fragment>
  <ForwardRef
    className="section-card"
    onClick={[Function]}
    to="/1"
  >
    <div
      className="section-card__wrapper"
    >
      <img
        alt=""
        className="section-card__img"
        src=""
      />
    </div>
  </ForwardRef>
  <span
    className="bookmark"
  >
    <input
      className="bookmark-input"
      defaultChecked={false}
      id={1}
      name={1}
      onClick={[Function]}
      type="checkbox"
    />
    <label
      className="bookmark-star"
      htmlFor={1}
    />
  </span>
</React.Fragment>
`;
